### ISSUE TITLE
Improvement: adding in options for the applied className and the titleSeparator

### ISSUE THREAD

**Issue by tearingItUp786** [rockchalkwushock/rehype-code-titles#237]
Hey, there friend, 

It would be super helpful if we could update the plugin to accept options such as `className` and `titleSeparator`. For example, a user (in this case me) might want to do something like `language:title=an-example-title` versus `language:an-example-title` in their input. 

Here's what the new interface could look like. I'm more than happy to make a PR as well b/c I had to make changes to your plugin for it to fit my use case  😊.

```ts
import { BuildVisitor, visit } from "unist-util-visit";
import type * as H from "hast";
import type { VisitorResult } from "unist-util-visit";

type Options = {
  className?: string;
  titleSeparator?: string;
};

function rehypeConfigurableCodeTitles() {
  return function optionsHof({
    className = "rehype-configurable-code-title",
    titleSeparator = ":",
  }: Options) {
    return function transformer(tree: H.Root) {
      const visitor: BuildVisitor<H.Root, "element"> = (
        node,
        index,
        parent
      ): VisitorResult => {
        // a changed implementation that leverages the options
      }

      visit(tree, "element", visitor);
    };
  };
}

export default rehypeConfigurableCodeTitles;
```

**Comment by rockchalkwushock** [comment]
Hey @tearingItUp786 I like the idea may I ask why the separator needed to be changed? 

If you are willing to put up a PR that ensures there is backwards compatibility for current users I would be more than happy to add the feature.

**Comment by tearingItUp786** [comment]
Yup, more than happy to make a PR! I want the option for the separator because I'm coming from [gatsby-remark-code-titles](https://www.gatsbyjs.com/plugins/gatsby-remark-code-titles/) and all my mdx files are using the `:title=` syntax for the separator and I quite like it! 

I'll throw something up right now!

**Comment by rockchalkwushock** [comment]
Closed by #240
